specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.ApiManagement/ApiManagement.tsp

```diff
@@ -130,6 +130,70 @@ model ApiManagementGatewayApiRemovedEventData {
   ...ApiGatewayApiBase;
 }
 
+/** Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.CircuitBreaker.Opened event. */
+model ApiManagementCircuitBreakerOpenedEventData {
+  /** Name of the backend for which the circuit has opened. */
+  backendName: string;
+
+  /** Information related to the circuit breaker configured on the backend. */
+  circuitBreaker: ApiManagementCircuitBreakerProperties;
+}
+
+/** Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.CircuitBreaker.Closed event. */
+model ApiManagementCircuitBreakerClosedEventData {
+  /** Name of the backend for which the circuit has closed. */
+  backendName: string;
+
+  /** Information related to the circuit breaker configured on the backend. */
+  circuitBreaker: ApiManagementCircuitBreakerProperties;
+}
+
+/** Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.GatewayTokenNearExpiry event. */
+model ApiManagementGatewayTokenNearExpiryEventData {
+  /** Information related to a given self-hosted gateway deployment. */
+  gatewayInfo: ApiManagementGatewayProperties;
+
+  /** Information related to a an expired gateway token for a self-hosted gateway deployment. */
+  tokenInfo: ApiManagementNearExpiryGatewayTokenProperties;
+}
+
+/** Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.GatewayTokenExpired event. */
+model ApiManagementGatewayTokenExpiredEventData {
+  /** Information related to a given self-hosted gateway deployment. */
+  gatewayInfo: ApiManagementGatewayProperties;
+
+  /** Information related to a an expired gateway token for a self-hosted gateway deployment. */
+  tokenInfo: ApiManagementExpiredGatewayTokenProperties;
+}
+
+/** Information related to the circuit breaker configured on the backend. */
+model ApiManagementCircuitBreakerProperties {
+  /** Overview of all configured rules and respective details. */
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "Type is Object in swagger, we cannot define full schema given customer owns the rule names"
+  rules: Record<unknown>;
```

Comment:
Object in swagger should be: `Record<{}>` in tsp.

Reply:
Thanks for figuring that out - Updated